### ISSUE TITLE
Fail to build from source on Fedora 24

### ISSUE THREAD

**Issue by epico** [sunpinyin/sunpinyin#58]
Fedora 24 uses gcc 6.0 to compile packages, but sunpinyin crashes when run tslmpack.
URL: https://bugzilla.redhat.com/show_bug.cgi?id=1308160

Please fix it, thanks!

**Comment by tchaikov** [comment]
@epico thanks for reporting this issue. #59 should address this issue, pending on review.

**Comment by epico** [comment]
Here are the new build log:
http://koji.fedoraproject.org/koji/taskinfo?taskID=13172742

It seems still have some other problem...

**Comment by tchaikov** [comment]
@epico seems the f24 is building sunpinyin using an ancient src package, probably you need revamp it?

for example (https://kojipkgs.fedoraproject.org//work/tasks/2745/13172745/build.log)
```
src/pinyin/segmentor.h: In member function 'bool IPySegmentor::TSegment::operator<(const IPySegmentor::TSegment&) const':
src/pinyin/segmentor.h:65:30: warning: self-comparison always evaluates to false [-Wtautological-compare]
                 return m_len < m_len;
                        ~~~~~~^~~~~~~
```
this warning (a bug actually) was fixed back to oct, 2014, see https://github.com/sunpinyin/sunpinyin/blame/master/src/pinyin/segmentor.h#L65.


```
Compressing bow values...16384 float values ==> 16384 valuestslmpack: src/slm/tslmpack/arpa_conv.cpp:84: unsigned int CArpaLeafConv::get_pr_index(float): Assertion `prit != m_pr_map.end()' failed.
```

and tslmpack is superseded by slmpack since 0fff1e78d9a409205e025736286838721a2ccbf8, maybe @CasperVector could fill you with more details.

**Comment by CasperVector** [comment]
I replaced `tslmpack` with `slmpack` to allow for separate updating of the SLM file (`lm_sc.3gm.arpa`, formerly `lm_sc.t3g.arpa`; the latter is threaded, hence the `t`) and `dict.utf8` as long as all words in the SLM are present in the dictionary, in the cost that one needs to run `slmthread` when generating the final binary SLM file, which is actually quite a cheap procedure.

**Comment by epico** [comment]
Sorry, my mistake. Built for Fedora rawhide now.
